aten/src/ATen/native/Distributions.cpp

```diff
@@ -600,4 +576,72 @@ Tensor multinomial(
   return result;
 }
 
-}} // namespace at::native
+} // namespace at::native
+
+namespace meta {
+
+// ==================================================== Normal ========================================================
+
+TORCH_META_FUNC2(normal, float_Tensor) (
+  double mean,
+  const Tensor& std,
+  c10::optional<Generator> gen
+) {
+  set_output(std.sizes(), std.options());
+}
+
+TORCH_META_FUNC2(normal, Tensor_float) (
+  const Tensor& mean,
+  double std,
+  c10::optional<Generator> gen
+) {
+  set_output(mean.sizes(), mean.options());
+}
+
+TORCH_META_FUNC2(normal, Tensor_Tensor) (
+  Tensor const& mean,
+  Tensor const& std,
+  c10::optional<Generator> gen
+) {
```

Comment:
Don't we need some checks here to make sure that `mean` and `std` have compatible dtypes? Or does this operation work with arbitrary dtypes?

Reply:
Oh, I see that the checks are currently done within the `normal_out_impl` function.
Those checks should be moved to the `TORCH_META_FUNC`s so that when meta tensors are fed to this operation they also get these checks.
To this end, perhaps the changes to the next PR in this stack could be moved to this one

Reply:
the reason the checks are in templates is because those are used by RNG tests. so i'm not sure it's a good idea to remove them if they can be accessed other than via the structured api. or is it the caller's responsibility to ensure they are calling the right thing (see `aten/src/ATen/test/cpu_rng_test.cpp`)?